### PR TITLE
[JBIDE-26562] OSServerLauncher: dont join with maven switching job

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by jkopriva** [comment]
@adietish It works for me as you have described in JIRA, switching profiles is blocked:
image

**Comment by adietish** [comment]
@jkopriva ok, thx for confirming. In the meantime I think that I found the solution with the help of @robstryker. Things now work for me. Can you please re-verify it before we freeze? That would be awesome.

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/ActivateMavenProfileJob.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	private Action action;
+
+	public ActivateMavenProfileJob(Action action, IProject project) {
+		super(NLS.bind("{0} maven profile \"openshift\" for project {1}", 
+				action == Action.ACTIVATE? "Activating" : "Deactivating", project.getName()));
+		this.action = action;
+		this.project = project;
+		// need to lock the whole workspace to prevent dead locks with other, concurrent
+		// jobs that escalate to the workspace lock.
+		setRule(ResourcesPlugin.getWorkspace().getRuleFactory().createRule(ResourcesPlugin.getWorkspace().getRoot()));
```
Comment (robstryker) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/ActivateMavenProfileJob.java]: IWorkspaceRoot and IContainer both implement ISchedulingRule 
so odds are you can replace
setRule(ResourcesPlugin.getWorkspace().getRuleFactory().createRule(ResourcesPlugin.getWorkspace().getRoot()));

with
setRule(ResourcesPlugin.getWorkspace().getRoot());

File: plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/ActivateMavenProfileJob.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2016-2018 Red Hat Inc..
```
Comment (jkopriva) [inline plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/ActivateMavenProfileJob.java]: Please update copyright year.

**Comment by jkopriva** [comment]
@adietish Good job, works for me too. Can you please also review sonar issues and my comment? Otherwise we can merge.

**Pull request merged**